configuration/multiple_kernels.rst

```diff
@@ -178,6 +178,31 @@ In order to solve this issue, add the following configuration to your kernel:
             # allows to use app/Resources/views/ templates in the ApiKernel
             "%kernel.root_dir%/../app/Resources/views": ~
 
+Running Tests Using a Different Kernel
+--------------------------------------
+
+In Symfony applications, functional tests extend by default from the
+:class:`Symfony\\Bundle\\FrameworkBundle\\Test\\WebTestCase` class. Inside that
+class, a method called ``getKernelClass()`` tries to find the class of the kernel
+to use to run the application during tests. The logic of this method does not
+support multiple kernel applications, so your tests won't use the right kernel.
```

Comment:
It should be sufficient to always reset the static `$class` property inside the `tearDown()` method:

```php
protected function tearDown()
{
    parent::tearDown();

    static::$class = null;
}
```

The `parent::tearDown()` call is important to make sure that the kernel is shut down after the test was run.